thrimshim/thrimshim/main.py

```diff
@@ -147,14 +130,46 @@ def update_row(ident, new_row):
 			sql.SQL("{} = {}").format(
 				sql.Identifier(column), sql.Placeholder(column),
 			) for column in new_row.keys()
-		))
+	))
 	result = database.query(conn, build_query, id=ident, **new_row)
 	if result.rowcount != 1:
 		return 'Video likely already published', 403	
 			
 	logging.info('Row {} updated to state {}'.format(ident, new_row['state']))
 	return ''
 
+
+@app.route('/thrimshim/manual-link/<uuid:ident>', methods=['POST'])
+def manual_link(ident):
+	"""Manually set a video_link"""
+	link = flask.request.json
+	conn = app.db_manager.get_conn()
+	results = database.query(conn, """
+		UPDATE events 
+		SET state='DONE', upload_location = 'Manual', video_link = %s
+		WHERE id = %s""", link, str(ident))
```

Comment:
this should only work when `state = 'UNEDITED' OR (state = 'DONE' AND upload_location = 'Manual')`, ie. either this is the first edit, or the previous edit was already a manual upload.
reset flow should cover everything else

Reply:
Ok, so in the cases when we need to replace a youtube upload that has been blocked we will need call both reset and manual-link